### PR TITLE
Imagery layers

### PR DESCRIPTION
**Pull request by nikhilmandlik**
Closes NIRVSSOMCT-271
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by charlesh88** [comment]
- I've added testing notes for the layers functionality specifically in 271 in the NIRVSSOMCT repo.

**Comment by shefalijoshi** [comment]
@nikhilmandlik There was another change by @michaelrogers that refactored the filters into a different component. Was that PR merged? If not, it should be merged first.

**Comment by michaelrogers** [comment]
> @nikhilmandlik There was another change by @michaelrogers that refactored the filters into a different component. Was that PR merged? If not, it should be merged first.

It was indeed merged here: https://github.com/nasa/openmct/pull/4736

That PR added the ImageControls.vue component.

**Comment by nikhilmandlik** [comment]
> Please consolidate the Filter settings refactor and re-request review.

Yup, its based of those changes from (@michaelrogers ) and your (@shefalijoshi) changes from image layer pr 😃

**Review by charlesh88 — APPROVED** [review]
This is awesome, looks great.

**Review by shefalijoshi — CHANGES_REQUESTED** [review]

**Comment by shefalijoshi** [comment]
Notebook snapshot fails with this error:
image

File: src/plugins/imagery/components/ImageControls.vue [hunk 1]
```diff
@@ -267,6 +266,13 @@ export default {
 
             const newScaleFactor = this.zoomFactor + (this.shiftPressed ? -ZOOM_STEP : ZOOM_STEP);
             this.zoomImage(newScaleFactor, e.clientX, e.clientY);
+        },
+        toggleLayerVisibility(index) {
+            this.$emit('toggleLayerVisibility', index);
+        },
+        updateFilterValues(filters) {
+            this.filters = filters;
+            this.notifyFiltersChanged();
```
[the remark this answers is not in the document]
Comment (nikhilmandlik) [reply src/plugins/imagery/components/ImageControls.vue:275]: also used in `handleResetFilters` so better to keep it as it is.

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
@nikhilmandlik Still seeing the CORS issue for snapshots

**Comment by nikhilmandlik** [comment]
> @nikhilmandlik Still seeing the CORS issue for snapshots

Yes thats expected and browser prevents snapshotting/capture of page from different domain.

**Review by shefalijoshi — COMMENTED** [review]
Please resolve conflicts. Looks good otherwise.

**Review by shefalijoshi — APPROVED** [review]

**Pull request merged**